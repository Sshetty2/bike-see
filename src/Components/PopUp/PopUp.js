import React from 'react';
import Login from '../../Containers/Login/Login';
// import Trips from '../../Containers/Trips/Trips';

const PopUp = (props) => {
  return (
    <div className='overlay-div'>
      <div onBlur={() => props.history.push('/')} className='login-pop-up'>
        <h3>{props.title}</h3>
        {props.title === 'User Login' && <Login />}
      </div>
    </div>
  )
}

export default PopUp;