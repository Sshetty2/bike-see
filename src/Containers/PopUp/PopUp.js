import React from 'react';
import Login from '../Login/Login';
import { connect } from 'react-redux';
import { setError } from '../../actions';
import PropTypes from 'prop-types';

export const PopUp = (props) => {
  const closePopUp = () => {
    const { history, setError } = props;
    setError('');
    history.replace('/');
  }

  return (
    <div className='overlay-div'>
      <div className='login-pop-up'>
        <i onClick={closePopUp} className="fas fa-times close-icon"></i>
        <h3>{props.title}</h3>
        {props.title === 'User Login' && <Login history={props.history} login={true}/>}
        {props.title === 'User Sign Up' && <Login history={props.history} login={false}/>}
      </div>
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  setError: (error) => dispatch(setError(error))
});

PopUp.propTypes = {
  setError: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(PopUp);