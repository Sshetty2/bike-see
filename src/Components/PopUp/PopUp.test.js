import React from 'react';
import PopUp from './PopUp';
import { shallow } from 'enzyme';

describe('PopUp', () => {
  let wrapper;
  let mockHistory;
  let mockTitle;

  beforeEach(() => {
    mockHistory = {replace: jest.fn()};
    mockTitle = 'User Login';
    wrapper = shallow(
      <PopUp 
        history={mockHistory}
        title={mockTitle}
      />
    )
  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.replace on click', () => {
    wrapper.find('.close-icon').simulate('click');
    expect(mockHistory.replace).toHaveBeenCalled();
  });
});