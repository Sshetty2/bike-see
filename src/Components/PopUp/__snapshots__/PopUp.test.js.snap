// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp should match the correct snapshot 1`] = `
<div
  className="overlay-div"
>
  <div
    className="login-pop-up"
  >
    <i
      className="fas fa-times close-icon"
      onClick={[Function]}
    />
    <h3>
      User Login
    </h3>
    <Connect(Login)
      history={
        Object {
          "replace": [MockFunction],
        }
      }
      login={true}
    />
  </div>
</div>
`;
